Use stable packages

greg0ire opened the pull request:
This will make it easier to contribute. It will also make reviews safer
since you will be able to tell if a PR is reponsible for a broken build
or not. Catching errors early is still possible by having a look at
allowed failures.

greg0ire committed:
Use stable packages

This will make it easier to contribute. It will also make reviews safer
since you will be able to tell if a PR is reponsible for a broken build
or not. Catching errors early is still possible by having a look at
allowed failures.

Majkl578 commented:
👍 See latest 8 PRs for reference - Symfony introduced a bug in YAML and broke our test suite (already fixed for next stable: symfony/symfony#25405).
Testing against dev dependencies is highly unreliable and non-deterministic.

greg0ire commented:
Why does Scrutinizer fail exactly?

Majkl578 commented:
According to documentation:
>```
> # Code Coverage decreased from previous inspection
> - 'project.metric_change("scrutinizer.test_coverage", < 0)'
> ```

Looks like some sort of a rounding bug to me.

lcobucci commented:
@greg0ire can you please rebase this branch to get the latest changes? TBH I like to run everything on with dev dependencies but I understand the frustration when you're trying to collaborate to a project and stuff doesn't work because weird stuff on dependencies. @Ocramius @guilhermeblanco any take on this?

BTW I've configured a cron job on travis to run the build on `master` daily, that should also help us to catch issues before of contributors.

lcobucci commented:
> Why does Scrutinizer fail exactly?

I've reported that, let's see what they say =)

Majkl578 commented:
@lcobucci Can we maybe add [conditional stage](https://blog.travis-ci.com/2017-09-12-build-stages-order-and-conditions) that will allow failure (or wouldn't execute) if the build is PR? So it would fail for master (and cron build), not for PR.

greg0ire commented:
Something is fishy, the build is entirely passing now. But unstable packages are still installed, so maybe the bug was fixed upstream?

greg0ire commented:
Hmmm... it's supposed to have been fixed 6 days ago, so prior to this bug. I don't understand this mystery. Maybe there was another bug in the meantime, fixed too?

greg0ire commented:
Or maybe is it the latest changes @lcobucci ?

greg0ire committed:
Implement conditional build

This will make cron builds fail, which is a bit more noticeable than an
allowed failure.

greg0ire commented:
Implemented @Majkl578 's proposal: the unstable build will "only" run for the `cron` build, which should be a bit more noticeable.

Majkl578 commented on .travis.yml:
```diff
@@ -56,8 +56,18 @@ jobs:
+        - composer config minimum-stability dev
+        - travis_retry composer update --prefer-dist
+
+    - stage: Lint
```
Looks like incorrect rebase, this shouldn't be here.

greg0ire replied:
Indeed, sorry

lcobucci reviewed (approved):
LGTM 👍

@Majkl578 do you have any comment?

Majkl578 reviewed (approved):
LGTM after recent changes. Maybe just add env DEV_DEPENDENCIES to distinguish the stage on overview in Travis.

greg0ire committed:
Add marker variable for Travis

greg0ire commented:
@Majkl578 done

lcobucci commented:
@greg0ire thanks!

![shipit](http://www.castlerock.com/images/news/SHIP1.jpg)
